src/System.Net.Sockets/src/System/Net/Sockets/UnixDomainSocketEndPoint.cs

```diff
@@ -0,0 +1,140 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Text;
+
+namespace System.Net.Sockets
+{
+    /// <summary>Represents a Unix Domain Socket endpoint as a path.</summary>
+    public sealed partial class UnixDomainSocketEndPoint : EndPoint
+    {
+        private const AddressFamily EndPointAddressFamily = AddressFamily.Unix;
+
+        private static readonly Encoding s_pathEncoding = Encoding.UTF8;
+        private static readonly Lazy<bool> s_udsSupported = new Lazy<bool>(() =>
+        {
+            try
+            {
+                new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified).Dispose();
+                return true;
+            }
+            catch
+            {
+                return false;
+            }
+        });
+
+        private readonly string _path;
+        private readonly byte[] _encodedPath;
+
+        public UnixDomainSocketEndPoint(string path)
+        {
+            if (path == null)
+            {
+                throw new ArgumentNullException(nameof(path));
+            }
+
+            // Pathname socket addresses should be null-terminated.
+            // Linux abstract socket addresses start with a zero byte, they must not be null-terminated.
+            bool isAbstract = IsAbstract(path);
```

Comment:
Does Windows support abstract socket addresses?  Will it fail later if one is passed in?

Reply:
It does.

Reply:
Oh, I see the test now.  Nice!

Reply:
Yes, Windows supports abstract addresses, but, it does not support the autobind feature available in linux (see man7 unix).